File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let res = 0, temp = 0;
    let left = 0, right = height.length-1;
    while(left < right){
        if (height[left] <= height[right]){
            temp = height[left]*(right-left);
            left++;
        }
        else {
            temp = height[right]*(right-left);
            right--;
        }
        if(res < temp) res = temp;
    }
    return res;
};

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area = 0, maxArea = 0;
        int left = 0, right = height.size()-1;
        while (left < right){
            if (height[left] <= height[right]){
                area = height[left]*(right-left);
                left++;
            }
            else {
                area = height[right]*(right-left);
                right--;
            }
            if (area > maxArea) maxArea = area;
        }
        return maxArea;
    }
};